File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityUnaryExpressionsTest.cpp
#include "ActivityUnaryExpressionsTest.h"
#include <alf/library/BasicInputOutput.hpp>
#include <alf/library/primitivebehaviors/BooleanFunctions.hpp>
#include <string>

void ActivityUnaryExpressionsTest::UnaryExpressionsTest()
{
  // Boolean
  bool b = false;
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(!b));

  // BitString
  int i = 123;
  alf::library::BasicInputOutput::WriteLine(std::to_string(static_cast<int>(~(i))));

  // Numeric
  alf::library::BasicInputOutput::WriteLine(std::to_string(+i));
  alf::library::BasicInputOutput::WriteLine(std::to_string(-i));
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityUnaryExpressionsTest.cs
/// <filename>
///     ActivityUnaryExpressionsTest.cs
/// </filename>
using Alf.Library;
using Alf.Library.PrimitiveBehaviors;
using System;

public class ActivityUnaryExpressionsTest
{
    public static void Execute()
    {
        // Boolean
        bool b = false;
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(!b));

        // BitString
        int i = 123;
        BasicInputOutput.WriteLine(Convert.ToString((int) ~(i)));

        // Numeric
        BasicInputOutput.WriteLine(Convert.ToString(+i));
        BasicInputOutput.WriteLine(Convert.ToString(-i));
    }

    public static void Main(string[] args)
    {
        ActivityUnaryExpressionsTest.Execute();
    }
}
